File: scripts/remove_auth.py
#!/usr/bin/env python3
import re
import os

files_to_update = [
    'apps/web/app/(dashboard)/dashboard/projects/[id]/versions/page.tsx',
    'apps/web/app/api/projects/[id]/versions/[versionId]/route.ts',
    'apps/web/app/api/projects/[id]/versions/route.ts',
    'apps/web/app/(dashboard)/dashboard/projects/[id]/pages/[pageId]/edit/page.tsx',
    'apps/web/app/(dashboard)/dashboard/projects/[id]/pages/new/page.tsx',
    'apps/web/app/(dashboard)/dashboard/projects/[id]/pages/page.tsx',
    'apps/web/app/(dashboard)/dashboard/projects/[id]/page.tsx',
    'apps/web/app/api/projects/[id]/pages/[pageId]/route.ts',
    'apps/web/app/api/projects/[id]/pages/route.ts',
    'apps/web/app/api/projects/[id]/route.ts',
]

for file_path in files_to_update:
    full_path = f'/home/user/clone-of-doc-tool/{file_path}'

    if not os.path.exists(full_path):
        print(f"Skipping {file_path} - not found")
        continue

    with open(full_path, 'r') as f:
        content = f.read()

    # Remove auth import
    content = re.sub(r"import \{ auth \} from '@/lib/auth';\n?", '', content)
    content = re.sub(r"import \{ redirect \} from 'next/navigation';\n?", '', content)

    # Add default user import if not present
    if 'getDefaultUser' not in content:
        # Find last import statement
        imports = list(re.finditer(r'^import .+?;\n', content, re.MULTILINE))
        if imports:
            last_import = imports[-1]
            insert_pos = last_import.end()
            content = content[:insert_pos] + "import { getDefaultUser } from '@/lib/default-user';\n" + content[insert_pos:]

    # Replace session = await auth() with user = getDefaultUser()
    content = re.sub(r'const session = await auth\(\);', 'const user = getDefaultUser();', content)

    # Remove auth checks
    content = re.sub(r'\s*if \(!session\?\.user\) \{\s*return NextResponse\.json\([^}]+\}, \{ status: \d+ \}\);\s*\}', '', content)
    content = re.sub(r'\s*if \(!session\) \{\s*return NextResponse\.json\([^}]+\}, \{ status: \d+ \}\);\s*\}', '', content)
    content = re.sub(r'\s*if \(!session\?\.user\) \{\s*redirect\([^)]+\);\s*\}', '', content)
    content = re.sub(r'\s*if \(!session\) \{\s*redirect\([^)]+\);\s*\}', '', content)

    # Replace session.user.id with user.id
    content = re.sub(r'session\.user\.id', 'user.id', content)

    # Replace session.user with user
    content = re.sub(r'session\.user', 'user', content)

    with open(full_path, 'w') as f:
        f.write(content)

    print(f"✓ Updated {file_path}")

print("\nDone!")

File: scripts/remove-auth.js
const fs = require('fs');
const path = require('path');
const glob = require('glob');

// Files to update
const patterns = [
  'apps/web/app/(dashboard)/**/*.tsx',
  'apps/web/app/api/**/*.ts',
  'apps/web/components/**/*.tsx'
];

function updateFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  let modified = false;

  // Remove auth imports
  if (content.includes("import { auth } from '@/lib/auth'")) {
    content = content.replace(/import { auth } from '@\/lib\/auth';\n?/g, '');
    modified = true;
  }

  if (content.includes("import { redirect } from 'next/navigation'") && content.includes('redirect(')) {
    // Keep redirect import but we'll remove usage
  }

  // Remove auth checks
  if (content.includes('const session = await auth()')) {
    content = content.replace(/const session = await auth\(\);\n?/g, '');
    modified = true;
  }

  // Remove redirect checks
  if (content.includes("if (!session?.user)") || content.includes("if (!session)")) {
    content = content.replace(/if \(!session\?\.user\) {\s*return NextResponse\.json\([^}]+\);\s*}\n?/g, '');
    content = content.replace(/if \(!session\?\.user\) {\s*redirect\([^)]+\);\s*}\n?/g, '');
    content = content.replace(/if \(!session\) {\s*return NextResponse\.json\([^}]+\);\s*}\n?/g, '');
    content = content.replace(/if \(!session\) {\s*redirect\([^)]+\);\s*}\n?/g, '');
    modified = true;
  }

  // Replace session.user with getDefaultUser()
  if (content.includes('session.user')) {
    if (!content.includes("import { getDefaultUser } from '@/lib/default-user'")) {
      // Add import after other imports
      const lastImport = content.lastIndexOf('\nimport');
      if (lastImport > -1) {
        const endOfLine = content.indexOf('\n', lastImport + 1);
        content = content.slice(0, endOfLine + 1) +
                  "import { getDefaultUser } from '@/lib/default-user';\n" +
                  content.slice(endOfLine + 1);
      }
    }

    // Add const user = getDefaultUser(); at start of function
    const functionStart = content.search(/export (?:default )?(?:async )?function/);
    if (functionStart > -1) {
      const openBrace = content.indexOf('{', functionStart);
      const nextLine = content.indexOf('\n', openBrace) + 1;
      if (!content.includes('const user = getDefaultUser()')) {
        content = content.slice(0, nextLine) +
                  '  const user = getDefaultUser();\n' +
                  content.slice(nextLine);
      }
    }

    content = content.replace(/session\.user/g, 'user');
    modified = true;
  }

  if (modified) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log(`Updated: ${filePath}`);
  }
}

// Process all files
patterns.forEach(pattern => {
  const files = glob.sync(pattern, { cwd: path.join(__dirname, '..') });
  files.forEach(file => {
    const fullPath = path.join(__dirname, '..', file);
    try {
      updateFile(fullPath);
    } catch (err) {
      console.error(`Error processing ${file}:`, err.message);
    }
  });
});

console.log('Done!');
